qiskit/compiler/assembler.py

```diff
@@ -272,7 +275,14 @@ def _parse_common_args(
             raise QiskitError(f"memory not supported by backend {backend_config.backend_name}")
 
         # try to set defaults for pulse, other leave as None
-        if backend_config.open_pulse:
+        pulse_param_set = (
+            qubit_lo_freq is not None
+            or meas_lo_freq is not None
+            or qubit_lo_range is not None
+            or meas_lo_range is not None
+            or schedule_los is not None
+        )
+        if pulse_qobj or backend_config.open_pulse and pulse_param_set:
```

Comment:
Maybe brackets on this `if` would help legibility - it's parsed as `pulse_qobj or (backend_config.open_pulse and pulse_param_set)` which seems correct to me, but I had to check the precedence tables to be sure.

Reply:
Sure I agree it's clearer with brackets that is the intended logic. I left it out because in the first local draft I had `pulse_param_set` inline instead as a separate variable and there were too many brackets. But pylint complained about the number of boolean expressions in the if so I split it out.

```suggestion
        if pulse_qobj or (backend_config.open_pulse and pulse_param_set):
```